### PR TITLE
Test improvements

### PR DESCRIPTION
**Pull request by tri-adam**
**Description of the Pull Request (PR):**

Improve Golang testing on the `development-3.0` branch. All go tests will now be run automatically by Travis.

In order to get a clean run, several modifications had to be made:
- Three source files in `src/runtime/startup/scontainer/` were moved into their own subdirectores, in order to avoid "main redeclared in this block" errors. In general, only one Go source file may have a `main()` in a given directory.
  - `mlocal/frags/Makefile.stub` was updated to reflect this restructuring
- `src/pkg/build/definition_parser_deffile_test.go` was refactored and test associated JSON files were updated to reflect the current `Definition` structure.
- Several other `fmt.Errorf()`/`fmt.Printf()`/`fmt.Println()` issues found by `go test` were addressed.

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [ ] I have tested this PR locally with a `make test`
- [x] This PR is NOT against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

**Comment by tri-adam** [comment]
@bauerm97, could use your expertise on this one. I _thought_ I'd set the right cgo flags (they're enough on my workstation) but Travis still isn't happy.

File: .travis.yml [hunk 1]
```diff
@@ -29,5 +29,7 @@ install:
   - sudo make -C ./buildtree install
 
 script:
+  - export CGO_CPPFLAGS="-I${GOPATH}/src/github.com/singularityware/singularity/builddir -I${GOPATH}/src/github.com/singularityware/singularity/src/runtime/c -I${GOPATH}/src/github.com/singularityware/singularity/src/runtime/c/lib"
```
Comment (bauerm97) [inline .travis.yml]: My expertise incoming:
`singularity/builddir` -> `singularity/buildtree`

File: src/cmd/singularity/cli/build.go [hunk 2]
```diff
@@ -59,25 +59,25 @@ var buildCmd = &cobra.Command{
 			// URI passed as arg[1]
 			def, err = build.NewDefinitionFromURI(args[1])
 			if err != nil {
-				fmt.Println("Unable to parse URI %s: ", args[1], err)
+				fmt.Printf("Unable to parse URI %s: %v\n", args[1], err)
```
Comment (bauerm97) [inline src/cmd/singularity/cli/build.go]: If you're making a change here, you could just use `sylog.Fatalf()` and remove the `os.Exit(1)` line. Just my 2c :-)

Comment (tri-adam) [reply src/cmd/singularity/cli/build.go]: Are we replacing all output with `sylog`? I was under the impression that console output would continue to be written directly?

File: .travis.yml (hunk 1 above)
Comment (tri-adam) [reply .travis.yml]: Ah, OK, as discussed, changing Travis to use the default option to `mconfig` since it's less confusing!

**Comment by tri-adam** [comment]
@bauerm97, think this is ready for a second pass if you're game...

**Review by bauerm97 — APPROVED** [review]
Looks good, Thanks!

**Pull request merged**